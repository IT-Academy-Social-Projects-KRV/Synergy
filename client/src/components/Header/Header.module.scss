.header {
    margin-bottom: 40px;
    color: #7872d6;
    font-weight: 500;
    display: flex;
    justify-content: space-between;
    align-items: center;
    .button {
        cursor: pointer;
        transition: all .5s ease;
        &:hover {
            transform: skewX(15deg);
        }
    }
    .exit {
        margin-right: 30px;
        img {
            margin-right: 5px;
        }
    }
}
.sideBlock {
    background: #fff;
    padding: 15px;
    flex: 0 0 15%;
    border-bottom-right-radius: 50px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    .logo_container {
        display: flex;
        align-items: center;
    }
}