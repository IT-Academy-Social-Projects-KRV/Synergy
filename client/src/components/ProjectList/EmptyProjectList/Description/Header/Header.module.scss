.text {
  font-weight: 700;
  font-size: 32px;
  color: #21143E;
}