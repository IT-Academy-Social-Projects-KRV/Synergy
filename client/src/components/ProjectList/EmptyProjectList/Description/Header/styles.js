const styles = {
  text: {
    fontWeight: 700,
    fontSize: '32px',
    color: '#21143E',
  },
};

export default styles;