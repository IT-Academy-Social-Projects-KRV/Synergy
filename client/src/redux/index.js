export * from './store';

export * from './actions/userAction';
export * from './actions/projectListAction';
export * from './actions/helperAction';
export * from './actions/itemAction';

export * from './reducers/userReducer';
export * from './reducers/projectListReducer';
export * from './reducers/helperReducer';
export * from './reducers/itemReducer';

export * from './selectors/selectors';

export * from './requests/projectListSortRequest';
export * from './requests/projectRequest';
export * from './requests/userRequest';
export * from './requests/itemRequest';