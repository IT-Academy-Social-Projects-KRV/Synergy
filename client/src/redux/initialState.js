
export const initialState = {
  user: {}, 
  projectList: [],
};