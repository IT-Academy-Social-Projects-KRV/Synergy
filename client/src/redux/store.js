import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './reducers/rootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware => 
    getDefaultMiddleware({ serializableCheck: false }),
});
