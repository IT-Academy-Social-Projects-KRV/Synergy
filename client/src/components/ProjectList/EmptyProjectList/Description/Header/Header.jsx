import { Typography } from '@mui/material';
import styles from './styles.js';

const Header = () => {
  return (
    <Typography sx={styles.text}>You not have any project.</Typography>
  );
};

export default Header;
