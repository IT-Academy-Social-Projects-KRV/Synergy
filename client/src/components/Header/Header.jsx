import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Header.module.scss';
import logo from '../../assets/images/HeaderImages/Logo_header.svg';
import button from '../../assets/images/HeaderImages/button.svg';
import exit from '../../assets/images/HeaderImages/exit.svg';
import { useDispatch } from 'react-redux';
import { setUser } from '../../redux';
import { Button } from '@mui/material';
import routes from '../../configs/routes';

const Header = ({ sideClass, changeClass }) => {

  const dispatch = useDispatch();

  const handleExit = () => {
    dispatch(setUser(null));
    sessionStorage.clear();
  };

  return (
    <header data-testid='headerBlock' className={styles.header}>
      <div data-testid='headerLogoBlock' className={styles.sideBlock + ' ' + sideClass}>
        <div className={styles.logo_container}>
          <img src={logo} alt='logo' />
          <span>Synergy | CRM</span>
        </div>
        <img
          data-testid='logoButton'
          src={button}
          className={styles.button}
          alt='logoButton'
          onClick={() => changeClass()}
        />
      </div>
      <div className={styles.exit}>
        <Link to={routes.NonAuthRoutes.pathToLogin}>
          <Button onClick={handleExit}>
            <img src={exit} alt='exit' />
            <span>Exit</span>
          </Button>
        </Link>
      </div>
    </header>
  );
};

export default Header;
