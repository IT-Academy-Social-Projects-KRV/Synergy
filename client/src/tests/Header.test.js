
import Header from '../components/Header/Header';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { setUser } from '../redux';

configure({ adapter: new Adapter() });

const mockStore = configureStore([]);

describe('Should render Header component with all items', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      sideClass: 'moved',
    });

    component = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on button click', () => {
    renderer.act(() => {
      component.root.findAllByType('div')[3].props.onClick();
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
      setUser(null)
    );
  });
});
