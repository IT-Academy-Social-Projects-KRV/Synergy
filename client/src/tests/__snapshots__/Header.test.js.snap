// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header component with all items should render with given state from Redux store 1`] = `
<header
  className="header"
  data-testid="headerBlock"
>
  <div
    className="sideBlock undefined"
    data-testid="headerLogoBlock"
  >
    <div
      className="logo_container"
    >
      <img
        alt="logo"
        src="Logo_header.svg"
      />
      <span>
        Synergy | CRM
      </span>
    </div>
    <img
      alt="logoButton"
      className="button"
      data-testid="logoButton"
      onClick={[Function]}
      src="button.svg"
    />
  </div>
  <div
    className="exit"
  >
    <div
      onClick={[Function]}
    >
      <img
        alt="exit"
        src="exit.svg"
      />
      <span>
        Exit
      </span>
    </div>
  </div>
</header>
`;
